src/dune_pkg/dev_tool.ml

```diff
@@ -5,30 +5,34 @@ type t =
   | Odoc
   | Ocamllsp
   | Utop
+  | Ocamlearlybird
 
 let to_dyn = function
   | Ocamlformat -> Dyn.variant "Ocamlformat" []
   | Odoc -> Dyn.variant "Odoc" []
   | Ocamllsp -> Dyn.variant "Ocamllsp" []
   | Utop -> Dyn.variant "Utop" []
+  | Ocamlearlybird -> Dyn.variant "Ocamlearlybird" []
 ;;
 
-let all = [ Ocamlformat; Odoc; Ocamllsp; Utop ]
+let all = [ Ocamlformat; Odoc; Ocamllsp; Utop; Ocamlearlybird ]
 
 let equal a b =
   match a, b with
   | Ocamlformat, Ocamlformat -> true
   | Odoc, Odoc -> true
   | Ocamllsp, Ocamllsp -> true
   | Utop, Utop -> true
-  | (Ocamlformat | Odoc | Ocamllsp | Utop), _ -> false
+  | Ocamlearlybird, Ocamlearlybird -> true
+  | (Ocamlformat | Odoc | Ocamllsp | Utop | Ocamlearlybird), _ -> false
```

Comment:
```suggestion
  | _,  _ -> false
```
I don't think there's much of a point in enumerating them all here, not in the `equal` function.

Reply:
FWIW these are both not the best way to write an `equal` function since adding a new constructor to the variant won't create an error. What you want is:
```
match a, b with
| Ocamlformat, Ocamlformat -> true
| OCamlformat, _ | _, OCamlformat -> false
...
```

That way it is always exhaustive and less prone to error.

Reply:
@Alizter in my original code I don't understand how adding a new constructor won't create an error. Can you explain please. I wrote it the current way precisely so that adding a new constructor will create an error because I spent half a day debugging a situation where exactly that happened.

Reply:
If you forget to add `| Ocamlformat, Ocamlformat -> true` to your way, you will not get a compiler error and it will be harder to spot. That's why in situations where we are adding many variants, catch all cases can be a bit of a foot gun.

The advantage of the way I proposed, although it can be a bit verbose, is that it keeps the concerns a bit more modular. When you add a constructor, you have to add two lines, rather than update existing code. Adding a constructor will always error, and its easier to add two lines rather than update two separate places.